src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/DatabaseHelper.cs

```diff
@@ -15,19 +17,151 @@ class DatabaseHelper
         /// <summary>
         /// Insert Customer record into table
         /// </summary>
-        internal static void InsertCustomerData(SqlConnection sqlConnection, string tableName, Customer customer)
+        internal static void InsertCustomerData(SqlConnection sqlConnection, SqlTransaction transaction, string tableName, Customer customer)
         {
-            using (SqlCommand sqlCommand = new SqlCommand(
+            using SqlCommand sqlCommand = new(
                 $"INSERT INTO [{tableName}] (CustomerId, FirstName, LastName) VALUES (@CustomerId, @FirstName, @LastName);",
                 connection: sqlConnection,
-                transaction: null,
-                columnEncryptionSetting: SqlCommandColumnEncryptionSetting.Enabled))
+                transaction: transaction,
+                columnEncryptionSetting: SqlCommandColumnEncryptionSetting.Enabled);
+
+            sqlCommand.Parameters.AddWithValue(@"CustomerId", customer.Id);
+            sqlCommand.Parameters.AddWithValue(@"FirstName", customer.FirstName);
+            sqlCommand.Parameters.AddWithValue(@"LastName", customer.LastName);
+            sqlCommand.ExecuteNonQuery();
+        }
+
+        /// <summary>
+        /// Validates that the results are the ones expected.
+        /// </summary>
+        /// <param name="sqlDataReader"></param>
+        public static void ValidateResultSet(SqlDataReader sqlDataReader)
+        {
+            // Validate the result set
+            int rowsFound = 0;
+
+            Assert.True(sqlDataReader.HasRows);
+            while (sqlDataReader.Read())
             {
-                sqlCommand.Parameters.AddWithValue(@"CustomerId", customer.Id);
-                sqlCommand.Parameters.AddWithValue(@"FirstName", customer.FirstName);
-                sqlCommand.Parameters.AddWithValue(@"LastName", customer.LastName);
+                if (sqlDataReader.FieldCount == 3)
+                {
+                    Assert.True(sqlDataReader.GetInt32(0) == 45, "Employee id didn't match.");
+                    Assert.True(sqlDataReader.GetString(1) == @"Microsoft", "Employee FirstName didn't match.");
+                    Assert.True(sqlDataReader.GetString(2) == @"Corporation", "Employee LastName didn't match.");
```

Comment:
NIT: Generally, my preference is using `Assert.Equal` instead of `Assert.True/False` for a better report on failures in each possible situation.

Reply:
```suggestion
                    Assert.Equal(45, sqlDataReader.GetInt32(0), "Employee id didn't match.");
                    Assert.Equal(@"Microsoft", sqlDataReader.GetString(1), "Employee FirstName didn't match.");
                    Assert.Equal(@"Corporation", sqlDataReader.GetString(2), "Employee LastName didn't match.");
```